Sources/Common/Core/Points/index.js

```diff
@@ -80,18 +80,20 @@ function vtkPoints(publicAPI, model) {
 // Object factory
 // ----------------------------------------------------------------------------
 
-const DEFAULT_VALUES = {
-  empty: true,
-  numberOfComponents: 3,
-  dataType: VtkDataTypes.FLOAT,
-  bounds: [1, -1, 1, -1, 1, -1],
-};
+function defaultValues(initialValues) {
+  return {
+    empty: true,
+    numberOfComponents: 3,
+    dataType: VtkDataTypes.FLOAT,
+    bounds: [1, -1, 1, -1, 1, -1],
+    ...initialValues,
+  };
+}
 
 // ----------------------------------------------------------------------------
 
 export function extend(publicAPI, model, initialValues = {}) {
-  Object.assign(model, DEFAULT_VALUES, initialValues);
-
+  Object.assign(model, defaultValues(model));
```

Comment:
I do not understand why you pass model as parameter to defaultValues

Reply:
It is in order that defaultValues of child classes are passed to parent class without being redundant with the passing of initialValues and risk being incoherent then when calling ``publicAPI.set(initialValues)` in new instance. However it will cause the setters of defaultValues not being called, so I'll update that.